diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4094b79..128e56b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,7 +76,6 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
-
 # Flags used for CHECK_CXX_SOURCE_COMPILES
 set(CMAKE_REQUIRED_FLAGS "-Wno-error")
 
@@ -104,6 +103,10 @@ endif()
 # base set of compile flags
 set (BASALT_CXX_FLAGS)
 
+if(WIN32)
+set (BASALT_CXX_FLAGS "/bigobj")
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
+
 # For avoiding libfmt >9 issues, see https://github.com/strasdat/Sophus/issues/366 #issuecomment-1229178088
 add_definitions(-DFMT_DEPRECATED_OSTREAM=1)
 

