diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3161667..1389fce 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,7 +60,9 @@ include(cmake/Flags.cmake)
 
 # Create library target
 add_library(${TARGET_NAME} ${XLINK_SOURCES})
-add_flag_source(src/shared/XLinkDevice.c "-Werror=switch-enum")
+if ("${CMAKE_C_COMPILER_ID}" MATCHES "^(AppleClang|Clang|GNU)$")
+    add_flag_source(src/shared/XLinkDevice.c "-Werror=switch-enum")
+endif()
 get_cmake_property(is_multi_config GENERATOR_IS_MULTI_CONFIG)
 if(is_multi_config)
     set_target_properties(${TARGET_NAME} PROPERTIES DEBUG_POSTFIX "d")

