diff --git a/cpr/session.cpp b/cpr/session.cpp
index b2263ec..c22b27b 100644
--- a/cpr/session.cpp
+++ b/cpr/session.cpp
@@ -6,6 +6,7 @@
 #include <cstdlib>
 #include <cstring>
 #include <fstream>
+#include <filesystem>
 #include <iostream>
 #include <memory>
 #include <optional>
@@ -134,6 +135,33 @@ Session::Session() : curl_(new CurlHolder()) {
     curl_easy_setopt(curl_->handle, CURLOPT_NOPROGRESS, 1L);
     curl_easy_setopt(curl_->handle, CURLOPT_ERRORBUFFER, curl_->error.data());
     curl_easy_setopt(curl_->handle, CURLOPT_COOKIEFILE, "");
+
+#ifdef __linux__
+    static const char* cert_path = nullptr;
+
+    // Find the system certificate store
+    if (cert_path == nullptr) {
+        // List of possible paths:
+        // https://github.com/cpp-pm/curl/blob/25d45e89d140d6ab27103cd7f8f6d7d6cf548d47/CMakeLists.txt#L919
+        static constexpr const char* certificatePaths[] = {
+                "/etc/ssl/certs/ca-certificates.crt", "/etc/pki/tls/certs/ca-bundle.crt",
+                "/usr/share/ssl/certs/ca-bundle.crt", "/usr/local/share/certs/ca-root-nss.crt",
+                "/etc/ssl/cert.pem"};
+
+        for (const auto& path : certificatePaths) {
+            if (std::filesystem::exists(path)) {
+                cert_path = path;
+                break;
+            }
+        }
+    }
+
+    // Set certificate path
+    if (cert_path != nullptr) {
+        curl_easy_setopt(curl_->handle, CURLOPT_CAINFO, cert_path);
+    }
+#endif
+
 #ifdef CPR_CURL_NOSIGNAL
     curl_easy_setopt(curl_->handle, CURLOPT_NOSIGNAL, 1L);
 #endif
